File: app/src/main/java/forpdateam/ru/forpda/ui/views/adapters/BaseViewHolder.java
package forpdateam.ru.forpda.ui.views.adapters;

import androidx.recyclerview.widget.RecyclerView;
import android.view.View;

/**
 * Created by radiationx on 14.09.17.
 */

public class BaseViewHolder<T> extends RecyclerView.ViewHolder {

    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    public void bind(T item, int section) {
    }

    public void bind(T item) {
    }

    public void bind(int position) {
    }

    public void bind() {
    }
}

File: app/src/main/java/forpdateam/ru/forpda/ui/views/adapters/BaseViewHolder.kt
package forpdateam.ru.forpda.ui.views.adapters

import android.view.View
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by radiationx on 14.09.17.
 */
open class BaseViewHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView) {
    open fun bind(item: T, section: Int) {
    }

    open fun bind(item: T) {
    }

    open fun bind(position: Int) {
    }

    fun bind() {
    }
}
